core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -74,6 +77,13 @@ private[spark] class CoarseGrainedExecutorBackend(
       val (hostname, _) = Utils.parseHostPort(hostPort)
       executor = new Executor(executorId, hostname, env, userClassPath, isLocal = false)
 
+    case NewTokens(tokens) =>
```

Comment:
This is not your fault, but this is horribly confusing. Why is a reply to the `RegisterExecutor` message, which is sent using `sendWithReply()`, received here?